tests/integration/features/test_lifecycle_no_overlay.py

```diff
@@ -0,0 +1,367 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import textwrap
+from pathlib import Path
+
+import pytest
+import yaml
+
+import craft_parts
+from craft_parts import Action, ActionProperties, ActionType, Features, Step
+
+basic_parts_yaml = textwrap.dedent(
+    """\
+    parts:
+      bar:
+        after: [foo]
+        plugin: nil
+
+      foo:
+        plugin: nil
+        source: a.tar.gz
+
+      foobar:
+        plugin: nil"""
+)
+
+
+@pytest.fixture(autouse=True)
+def _setup_module():
+    Features.reset()
+    Features(enable_overlay=False)
+    yield
+    Features.reset()
+
+
+def test_basic_lifecycle_actions(new_dir, mocker):
+    parts = yaml.safe_load(basic_parts_yaml)
+
+    Path("a.tar.gz").touch()
+
+    # no need to untar the file
+    mocker.patch("craft_parts.sources.tar_source.TarSource.provision")
+
+    # See https://gist.github.com/sergiusens/dcae19c301eb59e091f92ab29d7d03fc
```

Comment:
what's this?

Reply:
Oh I see that this is a version of another `test_basic_lifecycle_actions()` without overlays, right? Is the purpose just to make sure the basic lifecycle works without overlays? Can't we parametrize the original test instead?

Reply:
At the level we're testing, the results would be too different and parametrizing the original test would make it very confusing.